test/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter_test.cc

```diff
@@ -318,11 +323,34 @@ TEST_F(GrpcJsonTranscoderConfigTest, UnknownQueryParameterIsIgnored) {
   TranscoderInputStreamImpl request_in, response_in;
   TranscoderPtr transcoder;
   MethodInfoSharedPtr method_info;
-  const auto status =
-      config.createTranscoder(headers, request_in, response_in, transcoder, method_info);
+  UnknownVariableBindings unknown_variable_bindings;
+  const auto status = config.createTranscoder(headers, request_in, response_in, transcoder,
+                                              method_info, unknown_variable_bindings);
 
   EXPECT_TRUE(status.ok());
   EXPECT_TRUE(transcoder);
+  EXPECT_TRUE(unknown_variable_bindings.bindings().empty());
+}
+
+TEST_F(GrpcJsonTranscoderConfigTest, UnknownQueryParameterIsCaptured) {
+  auto proto_config = getProtoConfig(
+      TestEnvironment::runfilesPath("test/proto/bookstore.descriptor"), "bookstore.Bookstore");
+  proto_config.set_capture_unknown_query_parameters(true);
+  JsonTranscoderConfig config(proto_config, *api_);
+
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/shelves?foo=bar"}};
```

Comment:
can you test with multiple query params, and multiple params with the same name?
e.g. foo=bar+dsa&eep=baz&foo=asd?

just want to make sure we're thoughtful about the lack of normalization here

Reply:
Done, along with an updated API that does `key -> [values]` rather than `key -> value`